Formula/vulkan-tools.rb

```diff
@@ -0,0 +1,90 @@
+class VulkanTools < Formula
+  desc "Vulkan utilities and tools"
+  homepage "https://github.com/KhronosGroup/Vulkan-Tools"
+  url "https://github.com/KhronosGroup/Vulkan-Tools/archive/refs/tags/v1.3.248.tar.gz"
+  sha256 "9f06f13e588034e323df9222532b76518ace8bf5e4eb4ca9fbc6960addf03a93"
+  license "Apache-2.0"
+  head "https://github.com/KhronosGroup/Vulkan-Tools.git", branch: "main"
+
+  depends_on "cmake" => :build
+  depends_on "hatch" => :build
+  depends_on "python@3.11" => :build
+  depends_on "glslang"
+  depends_on "vulkan-headers"
+  depends_on "vulkan-loader"
+
+  on_macos do
+    depends_on "molten-vk"
+  end
+
+  on_linux do
+    depends_on "pkg-config" => :build
+    depends_on "libx11"
+    depends_on "libxcb"
+    depends_on "libxkbfile"
+    depends_on "libxrandr"
+    depends_on "wayland"
+    depends_on "wayland-protocols"
+  end
+
+  def install
+    if OS.mac?
+      # account for using already-built MoltenVK instead of the source repo
+      inreplace "mac_common.cmake",
+                "${MOLTENVK_DIR}/MoltenVK/icd/MoltenVK_icd.json",
+                "${MOLTENVK_DIR}/share/vulkan/icd.d/MoltenVK_icd.json"
+      inreplace buildpath.glob("*/macOS/*/*.cmake") do |s|
+        s.gsub! "${MOLTENVK_DIR}/MoltenVK/include",
+                "${MOLTENVK_DIR}/include"
+        s.gsub! "${MOLTENVK_DIR}/MoltenVK/dylib/macOS/libMoltenVK.dylib",
+                "${MOLTENVK_DIR}/lib/libMoltenVK.dylib"
+      end
+    end
+
+    args = [
+      "-DBUILD_ICD=ON",
+      "-DBUILD_CUBE=ON",
+      "-DBUILD_VULKANINFO=ON",
+      "-DINSTALL_ICD=OFF", # we will manually place it in a nonconflicting location
+      "-DGLSLANG_INSTALL_DIR=#{Formula["glslang"].prefix}",
+      "-DVULKAN_HEADERS_INSTALL_DIR=#{Formula["vulkan-headers"].prefix}",
+      "-DVULKAN_LOADER_INSTALL_DIR=#{Formula["vulkan-loader"].prefix}",
+    ]
+    args += if OS.mac?
+      ["-DMOLTENVK_REPO_ROOT=#{Formula["molten-vk"].prefix}"]
+    else
+      [
+        "-DBUILD_WSI_DIRECTFB_SUPPORT=OFF",
+        "-DBUILD_WSI_WAYLAND_SUPPORT=ON",
+        "-DBUILD_WSI_XCB_SUPPORT=ON",
+        "-DBUILD_WSI_XLIB_SUPPORT=ON",
+      ]
+    end
+    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    (prefix/"mock_icd").install (buildpath/"build/icd/VkICD_mock_icd.json").realpath,
+                                shared_library("build/icd/libVkICD_mock_icd")
+
+    return unless OS.mac?
+
+    bin.install_symlink prefix/"vulkaninfo/vulkaninfo"
+    (bin/"vkcube").write_env_script "/usr/bin/open", "-a #{prefix/"cube/vkcube.app"}", {}
+    (bin/"vkcubepp").write_env_script "/usr/bin/open", "-a #{prefix/"cube/vkcubepp.app"}", {}
```

Comment:
Does this work?
```suggestion
    (bin/"vkcube").write_exec_script prefix/"cube/vkcube.app/Contents/MacOS/vkcube"
    (bin/"vkcubepp").write_exec_script prefix/"cube/vkcubepp.app/Contents/MacOS/vkcubepp"
```

Reply:
Sadly not; the applications fail to start unless you launch them *as* applications, I think because of failing to find the bundled frameworks.

Reply:
Oops, my suggestion was wrong, try this:
```suggestion
    bin.write_exec_script prefix/"cube/vkcube.app/Contents/MacOS/vkcube"
    bin.write_exec_script prefix/"cube/vkcubepp.app/Contents/MacOS/vkcubepp"
```

Reply:
Ah, I didn't even try your suggestion as written because I'd already tried the correct one before.  That's how I arrived at the solution I did.

Reply:
Ok, that's fine then!